packages/cadl-ranch-specs/http/azure/resource-manager/models/resources/extension.tsp

```diff
@@ -0,0 +1,174 @@
+import "@typespec/http";
+import "@typespec/rest";
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-azure-resource-manager";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using Azure.Core;
+using Azure.ResourceManager;
+using TypeSpec.OpenAPI;
+
+namespace Azure.ResourceManager.Models.Resources;
+
+@resource("roleAssignments")
+model RoleAssignment is ExtensionResource<RoleAssignmentProperties> {
+  ...ResourceNameParameter<RoleAssignment>;
+}
+
+/** RoleAssignment properties */
+model RoleAssignmentProperties {
+  @doc("The description of the resource.")
+  description?: string;
+
+  /** The status of the last operation. */
+  @visibility("read")
+  provisioningState?: ProvisioningState;
+}
+
+@armResourceOperations
+interface RoleAssignments {
+  @scenario
+  @scenarioDoc("""
+    Resource GET operation.
+    Expected path: /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/test-rg/providers/Azure.ResourceManager.Models.Resources/roleAssignments/role
+    Expected query parameter: api-version=2023-12-01-preview
+    
+    Expected response body:
+    ```json
+    {
+      "id": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/test-rg/providers/Azure.ResourceManager.Models.Resources/roleAssignments/role",
+      "name": "role",
+      "type": "Azure.ResourceManager.Models.Resources/roleAssignments",
+      "properties":{
+        "description": "valid",
+        "provisioningState": "Succeeded"
+      },
+      "systemData": {
+        "createdBy": "AzureSDK",
+        "createdByType": "User",
+        "createdAt": <any date>,
+        "lastModifiedBy": "AzureSDK",
+        "lastModifiedAt": <any date>,
+        "lastModifiedByType": "User",
+      }
+    }
+    ```
+    """)
+  get is ArmResourceRead<RoleAssignment>;
```

Comment:
does the fact that it's a `RoleAssignment` object matter? With the scenario names we're really trying to convey what exactly we're testing. Right now, we're clearly testing `RoleAssignment`, which isn't a model from mgmt core, so I just want to make sure it's necessary that the scenario name includes `RoleAssignment`

Reply:
It's defined by us and just an `ExtensionResource`: https://github.com/Azure/cadl-ranch/pull/701/files#diff-caceefb500557a9f283bd4399e6ecc73278b9008e9b10b129c62403ea81a76cfR15

Agree that the name should be more generic than `RoleAssignment`. @v-hongli1 Please rename it to something else, e.g. `CustomExtensionResource`.

Reply:
Fixed in the new version.